core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the questions tab.
+ */
+
+require(
+  'components/question-directives/questions-list/' +
+  'questions-list.directive.ts');
+
+require('domain/utilities/url-interpolation.service.ts');
+require(
+  'domain/topics_and_skills_dashboard/' +
+  'topics-and-skills-dashboard-backend-api.service.ts');
+require('pages/topic-editor-page/services/topic-editor-state.service.ts');
+require('services/questions-list.service.ts');
+require('services/stateful/focus-manager.service.ts');
+import { Subscription } from 'rxjs';
+
+angular.module('oppia').component('questionsTab', {
+  template: require('./topic-questions-tab.component.html'),
+  controller: [
+    '$scope', '$window', 'FocusManagerService', 'QuestionsListService',
+    'TopicEditorStateService', 'TopicsAndSkillsDashboardBackendApiService',
+    function(
+        $scope, $window, FocusManagerService, QuestionsListService,
+        TopicEditorStateService, TopicsAndSkillsDashboardBackendApiService) {
+      var ctrl = this;
+      ctrl.directiveSubscriptions = new Subscription();
+      $scope.getGroupedSkillSummaries =
```

Comment:
Should this stuff be done in $onInit? Checking with @srijanreddy98 because I'm not sure.

Reply:
Technically, yes. But in the previous code it was outside onInit too. When migrating we move these to the ngOnInit function. I think its fine since this is just a rename.